Obsidian/WorldData/Generators/Overworld/ChunkBuilder.cs

```diff
@@ -144,9 +267,36 @@ internal static void CavesAndOres(GenHelper helper, Chunk chunk)
                     {
                         var oreNoise1 = helper.Noise.Ore(i).GetValue(worldX, y, worldZ);
                         var oreNoise2 = helper.Noise.Ore(i + ores.Length).GetValue(worldX, y, worldZ);
-                        if (oreNoise1 > 1.0 - OreSize && oreNoise2 > 1.0 - OreSize)
+                        if (oreNoise1 > 1.0 - OreSize && oreNoise2 > 1.0 - OreSize && GenerateOreCheck(y, i))
                         {
-                            chunk.SetBlock(worldX, y, worldZ, ores[i]);
+                            if (y <= 0 && y > -64)
```

Comment:
Something you might consider is to use a ternary in SetBock like so:
```csharp
chunk.SetBlock(worldX, y, worldZ, y > 0 ? ores[i] : deepores[i])
```
then you can avoid this level of if/else block